### PR TITLE
API-65: add integration tests on PQB [metric]

### PR DESCRIPTION
**Pull request by skeleton**
Add integration tests on metric PQB
(as it's quite boring to write test (and for you to read), I will create a dedicated PR for each attribute)
 
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/AbstractFilterTestCase.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @return mixed
+     */
+    protected function execute(array $filters)
+    {
+        $pqb = $this->get('pim_catalog.query.product_query_builder_factory')->create();
+
+        foreach ($filters as $filter) {
+            $context = isset($filter[3]) ? $filter[3] : [];
+            $pqb->addFilter($filter[0], $filter[1], $filter[2], $context);
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/AbstractFilterTestCase.php]: why not just http://php.net/manual/fr/function.call-user-func-array.php ?

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/Metric/LocalizableScopableFilterIntegration.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+ */
+class LocalizableScopableFilterIntegration extends AbstractFilterTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count) {
+            $this->createAttribute([
+                'code'                => 'a_metric_scopable_localizable',
```
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/Metric/LocalizableScopableFilterIntegration.php]: a_scopable_localizable_metric

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/Metric/ScopableFilterIntegration.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+ */
+class ScopableFilterIntegration extends AbstractFilterTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count) {
+            $this->createAttribute([
+                'code'                => 'a_metric_scopable',
```
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/Metric/ScopableFilterIntegration.php]: a_scopable_metric

File: src/Pim/Bundle/CatalogBundle/tests/integration/Doctrine/Filter/AbstractFilterTestCase.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        $this->get('pim_catalog.updater.attribute')->update($attribute, $data);
+        $this->get('pim_catalog.saver.attribute')->save($attribute);
+    }
+
+    /**
+     * @param array $filters
+     *
+     * @return mixed
+     */
+    protected function execute(array $filters)
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Doctrine/Filter/AbstractFilterTestCase.php]: as I execute the PQB, I keep like that

File: src/Pim/Bundle/CatalogBundle/tests/integration/Doctrine/Filter/AbstractFilterTestCase.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        return $pqb->execute();
+    }
+
+    /**
+     * @param CursorInterface $result
+     * @param array           $expected
+     */
+    protected function getResults($result, array $expected = [])
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Doctrine/Filter/AbstractFilterTestCase.php]: ok for `assert()` but I keep the second optional, it's not a mandatory to have an expected data

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/AbstractFilterTestCase.php (hunk 1 above)
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/AbstractFilterTestCase.php]: I could, but if I have to add new parameters in $filters, all tests will failed

**Pull request merged**